gluten-flink/planner/src/main/java/org/apache/gluten/rexnode/functions/BaseRexCallConverters.java

```diff
@@ -62,6 +62,14 @@ public DefaultRexCallConverter(String functionName) {
   public TypedExpr toTypedExpr(RexCall callNode, RexConversionContext context) {
     List<TypedExpr> params = getParams(callNode, context);
     Type resultType = getResultType(callNode);
+    // TODO: cast don't support input and result has same type. Refine it.
+    if (functionName.equals("cast")
+        && resultType
+            .getClass()
+            .getName()
+            .equals(params.get(0).getReturnType().getClass().getName())) {
```

Comment:
Maybe, just compare `resultType.getClass() == params.get(0).getReturnType().getClass()`.

Reply:
In the latest main branch, maybe, the code below can be removed if it's included here. 
```
if (sourceType instanceof TimestampType && resultType instanceof TimestampType) {
        return sourceExpr;
}
```

Reply:
done